sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -6193,11 +6193,19 @@ private <T> Flux<FeedResponse<T>> nonDocumentReadFeedInternal(
         };
 
         Function<RxDocumentServiceRequest, Mono<FeedResponse<T>>> executeFunc =
-            request -> readFeed(request)
-                .map(response -> feedResponseAccessor.createFeedResponse(
-                                    response,
-                                    DefaultCosmosItemSerializer.INTERNAL_DEFAULT_SERIALIZER,
-                                    klass));
+            request -> {
+                MetadataThrottlingRetryPolicy metadataThrottlingRetryPolicy = new MetadataThrottlingRetryPolicy(
```

Comment:
should I scope this change to only PartitionKeyRange? this method can be called by other resource types: getDatabases, readCollections, readTriggeres etc which all falls under metadata category